diff --git a/src/parser.c b/src/parser.c
index 9b9bd8e..55b2f80 100644
--- a/src/parser.c
+++ b/src/parser.c
@@ -25,6 +25,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <limits.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/mman.h>
@@ -667,7 +668,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
             }
         }
 
-        if (parse_int(dec, -9223372036854775807L - 1L, 9223372036854775807L, 10, &num, node->schema->name, line)
+        if (parse_int(dec, LLONG_MIN, LLONG_MAX, 10, &num, node->schema->name, line)
                 || validate_length_range(2, 0, 0, ((long double)num)/(1 << type->info.dec64.dig), stype,
                                          node->value_str, line)) {
             return EXIT_FAILURE;
@@ -785,7 +786,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_INT8:
-        if (parse_int(node->value_str, -128, 127, 0, &num, node->schema->name, line)
+        if (parse_int(node->value_str, SCHAR_MIN, SCHAR_MAX, 0, &num, node->schema->name, line)
                 || validate_length_range(1, 0, num, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -793,7 +794,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_INT16:
-        if (parse_int(node->value_str, -32768, 32767, 0, &num, node->schema->name, line)
+        if (parse_int(node->value_str, SHRT_MIN, SHRT_MAX, 0, &num, node->schema->name, line)
                 || validate_length_range(1, 0, num, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -801,7 +802,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_INT32:
-        if (parse_int(node->value_str, -2147483648, 2147483647, 0, &num, node->schema->name, line)
+        if (parse_int(node->value_str, INT_MIN, INT_MAX, 0, &num, node->schema->name, line)
                 || validate_length_range(1, 0, num, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -809,7 +810,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_INT64:
-        if (parse_int(node->value_str, -9223372036854775807L - 1L, 9223372036854775807L, 0, &num,
+        if (parse_int(node->value_str, LLONG_MIN, LLONG_MAX, 0, &num,
                 node->schema->name, line)
                 || validate_length_range(1, 0, num, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
@@ -818,7 +819,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_UINT8:
-        if (parse_uint(node->value_str, 255, 0, &unum, node->schema->name, line)
+        if (parse_uint(node->value_str, UCHAR_MAX, 0, &unum, node->schema->name, line)
                 || validate_length_range(0, unum, 0, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -826,7 +827,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_UINT16:
-        if (parse_uint(node->value_str, 65535, 0, &unum, node->schema->name, line)
+        if (parse_uint(node->value_str, USHRT_MAX , 0, &unum, node->schema->name, line)
                 || validate_length_range(0, unum, 0, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -834,7 +835,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_UINT32:
-        if (parse_uint(node->value_str, 4294967295, 0, &unum, node->schema->name, line)
+        if (parse_uint(node->value_str, UINT_MAX, 0, &unum, node->schema->name, line)
                 || validate_length_range(0, unum, 0, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -842,7 +843,7 @@ lyp_parse_value(struct lyd_node_leaf_list *node, struct lys_type *stype, int res
         break;
 
     case LY_TYPE_UINT64:
-        if (parse_uint(node->value_str, 18446744073709551615UL, 0, &unum, node->schema->name, line)
+        if (parse_uint(node->value_str, ULLONG_MAX, 0, &unum, node->schema->name, line)
                 || validate_length_range(0, unum, 0, 0, stype, node->value_str, line)) {
             return EXIT_FAILURE;
         }
@@ -887,4 +888,4 @@ lyp_get_next_union_type(struct lys_type *type, struct lys_type *prev_type, int *
     }
 
     return ret;
-}
\ No newline at end of file
+}
